prov/efa/src/efa_hmem.c

```diff
@@ -50,7 +50,22 @@ static size_t efa_max_eager_msg_size_with_largest_header(struct efa_domain *efa_
 }
 #endif
 
-static int efa_hmem_info_init_protocol_thresholds(struct efa_hmem_info *info, enum fi_hmem_iface iface, struct efa_domain *efa_domain) {
+/**
+ * @brief  Initialize the various protocol thresholds tracked in efa_hmem_info
+ *         according to the given FI_HMEM interface. This function is intentionally
+ *         designed to allow a given efa_hmem_info struct to be intitialized as
+ *         specified by any FI_HMEM interface supported by EFA. As such, it's
+ *         necessary to specify the FI_HMEM interface and efa_hmem_info pointer
+ *         separately.
+ *
+ * @param[in]   iface       The FI_HMEM interface to adhere to
+ * @param[in]   efa_domain  Pointer to struct efa_domain
+ * @param[out]  info        Pointer to efa_hmem_info struct to initialize
+ *
+ * @return      0
+ */
+static int efa_hmem_info_init_protocol_thresholds(enum fi_hmem_iface iface, struct efa_domain *efa_domain, struct efa_hmem_info *info)
```

Comment:
I do not think you need the `info` argument, as it can be assessed by `domain->hmem_info[iface]`

Reply:
I think function name should have `domain` in it because the function operates on `domain`. so `efa_domain_hmem_info_init_protocol_thresholds`

Reply:
> I do not think you need the `info` argument, as it can be assessed by `domain->hmem_info[iface]`

Not entirely. For `FI_HMEM_CUDA` when p2p is unavailable, [we initialize using `FI_HMEM_SYSTEM` default thresholds](https://github.com/ofiwg/libfabric/blob/9873561750112016ef69a7b0fc6324978d780043/prov/efa/src/efa_hmem.c#L150-L151). So we can't simply deduce the appropriate `efa_hmem_info` struct from the `iface` alone.

Reply:
Would  it make sense to handle the logic in `efa_hmem_info_init_protocol_thresholds`? e.g `efa_hmem_info_init_cuda` set `hmem_info->p2p_supported_by_device`. Then `efa_hmem_info_init_protocol_thresholds` decide threshold based on value of  `hmem_info->p2p_supported_by_device`

Reply:
Yeah, I think that makes more sense.